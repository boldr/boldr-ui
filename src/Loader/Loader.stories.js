import React from 'react';
import { storiesOf } from '@storybook/react';

import { action } from '@storybook/addon-actions';

import Loader from './Loader';

storiesOf('Loader', module).add('default', () => <Loader />);
