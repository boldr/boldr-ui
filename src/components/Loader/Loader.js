/* @flow */
import React from 'react';
import classnames from 'classnames';
import { StyleClasses, BOLDR_NS } from '../../theme/styleClasses';

type Props = {
  className: string,
};

const BASE_ELEMENT = StyleClasses.LOADER;

function Loader(props: Props) {
  const classes = classnames(BASE_ELEMENT, props.className);
  return (
    <div className={classes}>
      <div className={`${BASE_ELEMENT}__item`}>
        <div />
        <div />
        <div />
      </div>
    </div>
  );
}

export default Loader;
