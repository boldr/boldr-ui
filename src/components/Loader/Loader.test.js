import React from 'react';
import { createStore } from 'redux';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Loader from './Loader';

describe('<Loader />', () => {
  const wrapper = shallow(<Loader />);
  it('renders <Loader /> without breaking', () => {
    expect(wrapper.find('div').length).toBe(5);
  });
  it('renders snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
