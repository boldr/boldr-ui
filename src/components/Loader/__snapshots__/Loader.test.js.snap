// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> renders snapshot 1`] = `
<div
  className="boldrui-loader"
>
  <div
    className="boldrui-loader__item"
  >
    <div />
    <div />
    <div />
  </div>
</div>
`;
